sql-tumble-window-test/src/main/java/com/hazelcast/jet/sql/tests/tumblewindow/AbstractTumbleWindowTest.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.tests.tumblewindow;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.util.UuidUtil;
+import com.hazelcast.jet.sql.impl.connector.kafka.KafkaSqlConnector;
+import com.hazelcast.jet.tests.common.AbstractSoakTest;
+import com.hazelcast.jet.tests.common.Util;
+import com.hazelcast.jet.tests.common.sql.TestRecordProducer;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.sql.HazelcastSqlException;
+import com.hazelcast.sql.SqlResult;
+import com.hazelcast.sql.SqlRow;
+import com.hazelcast.sql.SqlService;
+
+import java.time.Duration;
+import java.util.Iterator;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.hazelcast.jet.sql.impl.connector.SqlConnector.OPTION_KEY_FORMAT;
+import static com.hazelcast.jet.sql.impl.connector.SqlConnector.OPTION_VALUE_FORMAT;
+
+public abstract class AbstractTumbleWindowTest extends AbstractSoakTest {
+    private static final int DEFAULT_QUERY_TIMEOUT_MILLIS = 100;
+    private static final int PROGRESS_PRINT_QUERIES_INTERVAL = 500;
+
+    private static final int EVENT_START_TIME = 0;
+    private static final int EVENT_WINDOW_COUNT = 10;
+    private static final int EVENT_TIME_INTERVAL = 1;
```

Comment:
If one adjusts this parameter, the assertions in `assertQuerySuccessful` will fail because they don't take it into account. Better to remove it?